src/Features/Core/Portable/InlineHints/AbstractInlineParameterNameHintsService.cs

```diff
@@ -32,21 +35,130 @@ public async Task<ImmutableArray<InlineParameterHint>> GetInlineParameterNameHin
+        {
+            // Only relevant if we have two or more parameters.
+            if (parameterHints.Count <= 1)
+                return false;
+
+            return ParametersDifferOnlyByAlphaSuffix(parameterHints) ||
+                   ParametersDifferOnlyByNumericSuffix(parameterHints);
+
+            static bool ParametersDifferOnlyByAlphaSuffix(ArrayBuilder<InlineParameterHint> parameterHints)
+            {
+                if (!HasAlphaSuffix(parameterHints[0], out var firstPrefix))
+                    return false;
+
+                for (var i = 1; i < parameterHints.Count; i++)
+                {
+                    if (!HasAlphaSuffix(parameterHints[i], out var nextPrefix))
+                        return false;
+
+                    if (!firstPrefix.Span.Equals(nextPrefix.Span, StringComparison.Ordinal))
+                        return false;
+                }
+
+                return true;
+            }
+
+            static bool ParametersDifferOnlyByNumericSuffix(ArrayBuilder<InlineParameterHint> parameterHints)
+            {
+                if (!HasNumericSuffix(parameterHints[0], out var firstPrefix))
+                    return false;
+
+                for (var i = 1; i < parameterHints.Count; i++)
+                {
+                    if (!HasNumericSuffix(parameterHints[i], out var nextPrefix))
+                        return false;
+
+                    if (!firstPrefix.Span.Equals(nextPrefix.Span, StringComparison.Ordinal))
+                        return false;
+                }
+
+                return true;
+            }
+
+            static bool HasAlphaSuffix(InlineParameterHint hint, out ReadOnlyMemory<char> prefix)
+            {
+                var name = hint.Parameter?.Name;
+
+                // Has to end with A-Z
+                // That A-Z can't be following another A-Z (that's just a capitalized word).
+                if (name?.Length >= 2 &&
+                    IsUpperAlpha(name[^1]) &&
+                    !IsUpperAlpha(name[^2]))
+                {
+                    prefix = name.AsMemory()[..^1];
+                    return true;
+                }
+
+                prefix = default;
+                return false;
+            }
+
+            static bool HasNumericSuffix(InlineParameterHint hint, out ReadOnlyMemory<char> prefix)
+            {
+                var name = hint.Parameter?.Name;
+
+                // Has to end with 0-9.  only handles single-digit numeric suffix for now for simplicity
+                if (name?.Length >= 2 &&
+                    IsNumeric(name[^1]))
+                {
+                    prefix = name.AsMemory()[..^1];
+                    return true;
+                }
+
+                prefix = default;
+                return false;
+            }
+
+            static bool IsUpperAlpha(char c)
+                => c is >= 'A' and <= 'Z';
```

Comment:
Why not char.IsLetter() && char.IsUpper()? What about other languages?

Reply:
because, honestly, this is about how APIs commonly do things.  which means ending variables in A, B, C..., or 1, 2, 3.  If someone says they have an API that is in a foreign language, and they abide by some sort of 'suffix determination' we can reconsider this.  but i imagine it's a literal non-issue.